crates/redstone/src/helpers/slice.rs

```diff
@@ -0,0 +1,101 @@
+/// Performs lookup for repeated value in the list.
+///
+/// # Arguments
+///
+/// * `list` - A `&[T]` list where `T` implements `PartialEq`, `Eq`, `Copy` and `Ord` traits.
+///
+/// # Returns
+///
+/// Returns a `Option<T>` which contains first repeated element found in the list or if there is no repeated
+/// element then None otherwise.
+pub fn has_duplicates<T>(list: &[T]) -> Option<T>
+where
+    T: PartialEq + Eq + Copy + Ord,
+{
+    if list.len() < 2 {
+        return None;
+    }
+    if list.len() == 2 {
+        if list[0] == list[1] {
+            return Some(list[0]);
+        }
+        return None;
+    }
+    let mut list: Vec<T> = list.to_vec();
+    list.sort();
```

Comment:
minor: sort_unstable will be slightly faster

Reply:
Good point. Will do.